File: python/key/align_money.py
"""
Name: align_money.py
This demo program is designed to:
	Demonstrate how to align multiple columns of numbers (money
	in this case) so they are pleasing to the eye
Author:
	KEY
Source:
	Elijah Harrison
	James Helfrich
"""

"""
Import Statements/Globals
"""

"""
main()...
"""
def align_money():
	"""
	Demonstrates how to align multiple columns of numbers (money in this case)
	so they are pleasing to the eye.
	"""

	# Format the output for money
	fixed = lambda num: f"{num : .2f}" # Format to two decimals and right-align

	# Display the first row
	row1 = ("$" +
		f"{fixed(124.45) : >7}" +
		"\t$" +
		f"{fixed(321.31) : >7}" +
		"\n")
	print(row1, end="")

	# Display the second row
	row2 = ("$" +
		f"{fixed(1.74) : >7}" +
		"\t$" +
		f"{fixed(4.21) : >7}" +
		"\n")
	print(row2, end="")

	# Display the third row
	row3 = ("$" +
		f"{fixed(7539.12) : >7}" +
		"\t$" +
		f"{fixed(54.92) : >7}" +
		"\n")
	print(row3, end="")

if __name__ == "__main__":
	align_money()

File: cpp/Unit-1/exercise11-alignMoney/alignMoney.cpp
/***********************************************************************
 * This demo program is designed to:
 *      Demonstrate how to align multiple columns of numbers (money
 *      in this case) so they are pleasing to the eye
 ************************************************************************/

#include <iostream>
#include <iomanip>
using namespace std;

/**********************************************************************
 * main()...
 ***********************************************************************/
int main()
{
   // format the output for money
   cout.setf(ios::fixed);     // no scientific notation
   cout.setf(ios::showpoint); // always show the decimal point
   cout.precision(2);         // unless you own a gas station, money
                              //    is formatted to two decimals
   
   // display the first row
   cout << "$"
        << setw(7) << 124.45
        << "\t$"
        << setw(7) << 321.31
        << endl;
   
   // display the second row
   cout << "$"
        << setw(7) << 1.74
        << "\t$"
        << setw(7) << 4.21
        << endl;

   // display the third row
   cout << "$"
        << setw(7) << 7539.12
        << "\t$"
        << setw(7) << 54.92
        << endl;

   return 0;
}
